numpy/lib/tests/test_index_tricks.py

```diff
@@ -14,6 +16,33 @@
 class TestRavelUnravelIndex(object):
     def test_basic(self):
         assert_equal(np.unravel_index(2, (2, 2)), (1, 0))
+
+        # test backwards compatibility with older dims
+        # keyword argument; see Issue #10586
+        with pytest.warns(DeprecationWarning):
+            # we should achieve the correct result
+            # AND raise the appropriate warning
+            # when using older "dims" kw argument
+            assert_equal(np.unravel_index(indices=2,
+                                          dims=(2, 2)),
+                                          (1, 0))
+
+        # test that new shape argument works properly
+        assert_equal(np.unravel_index(indices=2,
+                                      shape=(2, 2)),
+                                      (1, 0))
+
+        # test that an invalid second keyword argument
+        # is properly handled
+        with pytest.raises(TypeError):
+            np.unravel_index(indices=2, hape=(2, 2))
+
+        with pytest.raises(TypeError):
+            np.unravel_index(2, hape=(2, 2))
+
+        with pytest.raises(TypeError):
+            np.unravel_index(254, ims=(17, 94))
+
```

Comment:
We should still use the test-framework agnostic suppress_warnings, assert_raises instead of pytest-specific tests

Reply:
Is there a specific reason for that? Still trying to retain compatibility with other frameworks at this point or just want to maintain thematic consistency across the testsuite for now?

Reply:
@charris - should we be preferring `assert_raises` or are we ready to commit to pytest?